backend/src/main/java/org/umcs/appollo/services/PollService.java

```diff
@@ -0,0 +1,73 @@
+package org.umcs.appollo.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import org.umcs.appollo.converters.PollConverter;
+import org.umcs.appollo.model.PollEntity;
+import org.umcs.appollo.model.QuestionEntity;
+import org.umcs.appollo.model.api.Poll;
+import org.umcs.appollo.repository.PollRepository;
+import org.umcs.appollo.repository.QuestionRepository;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class PollService {
+    private final PollRepository pollRepository;
+    private final PollConverter pollConverter;
+    private final QuestionRepository questionRepository;
+
+    public PollService(PollRepository pollRepository, PollConverter pollConverter, QuestionRepository questionRepository) {
+        this.pollRepository = pollRepository;
+        this.pollConverter = pollConverter;
+        this.questionRepository = questionRepository;
+    }
+
+    public List<Poll> getPolls() {
+        List<Poll> polls = pollRepository.findAll()
+            .stream()
+            .map(pollConverter::FromEntityToApi)
+            .collect(Collectors.toList());
+        return polls;
+    }
+
+    public Poll getPoll(Integer id) {
+        Optional<PollEntity> result = pollRepository.findById(id);
+        if(result.isEmpty())
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Poll with id " + id + " not found.");
+
+        PollEntity pollEntity = result.get();
+        return pollConverter.FromEntityToApi(pollEntity);
+    }
+
+    public PollEntity createPoll(Poll poll) {
+        PollEntity pollEntity = pollConverter.FromApiToEntity(poll);
+        if(pollEntity==null) throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Poll not found");
```

Comment:
Tu raczej przy `createPoll` jeśli `pollEntity` jest `null`, to znaczy że `poll` było `null`, a to z kolei znaczy że było złe zapytanie, więc do wrócenia `400 Bad Request` zamiast tego.